### ISSUE TITLE
False Positive CA2016 for EnumeratorCancellationAttribute parameter

### ISSUE THREAD

**Issue by lambdageek** [dotnet/roslyn-analyzers#4452]
### Analyzer

**Diagnostic ID**: [CA2016](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2013): `Forward the CancellationToken parameter to methods that take one`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

Methods that return an `IAsyncEnumerable<T>` take a `[EnumeratorCancellation] CancellationToken ct = default` parameter but the `CancellationToken`  is passed to the async enumerable using the [`WithCancellation` extension method](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskasyncenumerableextensions.withcancellation?view=dotnet-plat-ext-5.0#System_Threading_Tasks_TaskAsyncEnumerableExtensions_WithCancellation__1_System_Collections_Generic_IAsyncEnumerable___0__System_Threading_CancellationToken_) rather than directly.

### Steps To Reproduce

Create a project with the following source file and enable analyzers:

```csharp
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;
using System.Threading;
using System.Threading.Tasks;

namespace Example {
    public static class FalsePositiveCA2016 {
         public static async Task<int> SumUp (CancellationToken ct = default) {
            int sum = 0;
            await foreach (int n in DelayNumbers (TimeSpan.FromMilliseconds(250)).WithCancellation (ct)) {
                //    CA2016 here:  ^^^^^^^^^^^^
                sum += n;
            }
            return sum;
        }

        public static async IAsyncEnumerable<int> DelayNumbers (TimeSpan delay, [EnumeratorCancellation] CancellationToken ct = default) {
            int j = 0;
            while (true) {
                try {
                    await Task.Delay(delay, ct);
                } catch (TaskCanceledException) {
                    yield break;
                }
                if (ct.IsCancellationRequested)
                    yield break;
                yield return j;
                ++j;
            }
        }
    }
}
```

### Expected behavior

No CA2016 warning

### Actual behavior

The call to `DelayNumbers` in the `async foreach` loop is highlighted with a CA2016 warning.

**Comment by lambdageek** [comment]
hm. not sure this is a false positive after all.
